### ISSUE TITLE
Improved playing queue

### ISSUE THREAD

**Issue by Bnyro** [libre-tube/LibreTube#1555]
### Describe your suggested feature

A visual interface for the already existing playing queue with support for removing items as well as changing the order of them by dragging and dropping.

The issue here is being opened to get potential ideas on how it should look UI-wise, I'll implement it myself in the end.
So, every kind of opinions or suggestions is welcome! :)

### Acknowledgements

- [X] I have searched the existing issues and this is a new ticket, **NOT** a duplicate or related to another open issue.
- [X] I have written a short but informative title.
- [X] I will fill out all of the requested information in this form.

**Comment by rancidfrog** [comment]
I see you removed Add to queue option from menu, as it was not working anyway

**Comment by GilesMunn** [comment]
Perhaps, if there's more than one video in the queue, an additional drop-down button "Queue" could appear above "Comments" in the video player? Then in that dropdown have the videos listed in the same style as the recommended section and let the user jump to a particular one by clicking on it?

**Comment by Bnyro** [comment]
My plans were to make it appear as a bottom sheet but not use the normal video or trending layout for it because it would lack the features of being able to reorder and remove the videos in the queue. It'd probably make sense to let users remove videos from the queue by swiping them to the left and moving them by holding and dragging.
I think the idea with the button above the comments is a good one ;)

**Comment by yhm4ps1jfp67nkv** [comment]
I think the queue should also work for the background player like in NewPipe, so I think this somewhat depends on #1113 and #1157

**Comment by Bnyro** [comment]
The queue already works with background play so those issues won't be a requirement for now :)
But you're right, the menu has to be reachable too while playing a video in the background.
